public_dropin_environments/java_codegen/Dockerfile

```diff
@@ -32,25 +40,13 @@ COPY --from=build /bin/mkdir /bin/mkdir
 # Required for custom-models to install dependencies
 COPY --from=build /usr/bin/pip /usr/bin/pip
 
-# Cleanup '__pycache__' directories. It solves an AsymmetricPrivateKey scanning error.
-COPY --from=build /usr/bin/rm /usr/bin/rm
-COPY --from=build /usr/bin/find /usr/bin/find
 
 # Just for convenience
 COPY --from=build /bin/ls /bin/ls
 
-
-COPY requirements.txt requirements.txt
-
 ENV VIRTUAL_ENV=/opt/venv
-
-RUN sh -c "python -m venv ${VIRTUAL_ENV} && \
-    . ${VIRTUAL_ENV}/bin/activate && \
-    python -m ensurepip --default-pip && \
-    python -m pip install --upgrade pip && \
-    python -m pip install --no-cache-dir -r requirements.txt && \
-    find ${VIRTUAL_ENV} -type d -name '__pycache__' -exec rm -rf {} + && \
-    rm -rf /usr/bin/rm /usr/bin/find"
+# Copy preinstalled virtualenv
+COPY --from=build --chown=nonroot ${VIRTUAL_ENV} ${VIRTUAL_ENV}
```

Comment:
this definitely needs testing with actual datarobot. we use 1000:1000 to run by default.